File: reimi/frontend/reimi_app/lib/domain/value_objects/blood_type.dart
import 'package:freezed_annotation/freezed_annotation.dart';

enum BloodType {
  @JsonValue('A')
  a,
  @JsonValue('B')
  b,
  @JsonValue('O')
  o,
  @JsonValue('AB')
  ab,
  @JsonValue('UNKNOWN')
  unknown,
}

File: reimi/backend/reimi_app/src/main/java/com/reimi/reimi_app/domain/model/profile/BloodType.java
package com.reimi.reimi_app.domain.model.profile;

public enum BloodType {

    A("A型"),
    B("B型"),
    O("O型"),
    AB("AB型"),
    UNKNOWN("わからない");

    private final String label;

    BloodType(String label) {
        this.label = label;
    }

    public String getLabel() {
        return label;
    }
}
